### PR TITLE
[Questions] Use Hexagonal Architecture

### PR DESCRIPTION
**Pull request by jeffrey-jian**
### Description

This pull request refactors the questions-service to adopt the **hexagonal architecture**, separating business logic from database concerns and improving testability. 

Question Hexagonal Architecture drawio (2)

### Changes made

- Created the `questions.repository.ts` abstract class
  - existing `questions.service.ts` now depends on this new abstract class to handle database operations
  
- Moved supabase-related methods to its own class: `questions.supabase.ts`
  - implements `questions.repository.ts`
 

### New directory structure
```text
src/
├── adapters/
│   └── controllers/
│   │   └── questions.controller.ts
│   └── db/
│       └── questions.supabase.ts
├── domains/
│   └── ports/
│       ├── questions.repository.ts
│       └── questions.service.ts
├── questions.module.ts
└── main.ts
```

### Future enhancements

**Testing**
- `nestjs` recommends using [automock](https://docs.nestjs.com/recipes/automock#introduction) library (now called suites) for testing, but there are issues with the library at the moment.
- I'll incorporate unit tests into our microservices once the this [issue](https://github.com/suites-dev/suites/issues/426) has been resolved (I've tried using their suggested workaround, it doesn't work)

### CODE REVIEW

**Comment by jeffrey-jian** [comment]
Merge #22 first, I'll rename the files from `questions.*` to `question.*` after.

File: project/apps/questions-service/src/adapters/controllers/questions.controller.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-import { QuestionsService } from './questions.service';
+import { QuestionsService } from 'src/domain/ports/questions.service';
 import {
   CreateQuestionDto,
   GetQuestionsQueryDto,
   UpdateQuestionDto,
 } from '@repo/dtos/questions';
+
 @Controller()
 export class QuestionsController {
```
Comment (jeffrey-jian) [inline project/apps/questions-service/src/adapters/controllers/questions.controller.ts:11]: Merge #22 first, I'll rename the files from `questions.*` to `question.*` after.

**Review by javinchua — APPROVED** [review]
lgtm, should the api-gateway structure be changed to use this approach as well?

**Comment by jeffrey-jian** [comment]
> lgtm, should the api-gateway structure be changed to use this approach as well?

yup, will look into updating api-gateway as well.

**Pull request merged**